src/rgw/rgw_rados.h

```diff
@@ -2427,6 +2427,16 @@ class RGWRados
   const string& get_current_period_id() {
     return current_period.get_id();
   }
+
+  int get_zonegroups_by_api(map<string, RGWZoneGroup>& zonegroups_by_api) {
+    int ret = -1;
+    if (!current_period.get_id().empty()) {
+      zonegroups_by_api = current_period.get_map().zonegroups_by_api;
+      ret = 0;
+    }
+    return ret;
+  }
+
```

Comment:
thanks. can we find a way to avoid copying the whole map here? it's especially expensive because each `RGWZoneGroup` itself has a few lists/maps. there are a few options, but the simplest is probably to narrow the interface to just search for a given api:
```c++
bool validate_zonegroup_api(const std::string& api) const {
```
what do you think?